enumerable.rb

```diff
@@ -0,0 +1,130 @@
+module Enumerable
+  def my_each
+    return to_enum(my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    rreturn to_enum(my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(my_select) unless block_given?
+
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_select

- [ ] my_select returns an enumerator if no block is given

- Yours is giving a SystemStackError
![image](https://user-images.githubusercontent.com/52209067/79289069-942dc880-7e85-11ea-9ce6-aaa042105d31.png)

- Real select
![image](https://user-images.githubusercontent.com/52209067/77681510-0bdd9700-6f5b-11ea-9340-ac116b7ed62e.png)

Reply:
done